### PR TITLE
Benchmark integration with the CI system

### PR DESCRIPTION
**Pull request by gflegar**
This PR adds jobs to the CI system which will run Ginkgo's benchmarks on a regular schedule (can be set up [here](https://gitlab.com/ginkgo-project/ginkgo-public-ci/pipeline_schedules), currently once a day, at 2am) and publish the results to ginkgo-project/ginkgo-data. This data can then be viewed in the [Benchmark explorer](https://ginkgo-project.github.io/ginkgo-benchmark-explorer/).

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -193,15 +217,124 @@ gh-pages:
...
+  only:
+    - schedules
+
+
+# Benchmark runs
+.benchmark_template: &default_benchmark
+  before_script: *default_benchmark_before_script
+  script:
+    - |
+      ${SSH_COMMAND} 'tee /dev/stderr | scl enable devtoolset-7 bash' \
```
Comment (hartwiganzt) [inline .gitlab-ci.yml]: Maybe a few more comments what is happening here?

File: benchmark/run_all_benchmarks.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    GROUP=$(${SSGET} -i $i -pgroup)
+    NAME=$(${SSGET} -i $i -pname)
+    RESULT_FILE="${RESULT_DIR}/${GROUP}/${NAME}.json"
+    mkdir -p $(dirname ${RESULT_FILE})
+    INPUT=$(${SSGET} -i $i -e)
+    echo "[{\"filename\": \"${INPUT}\", \"problem\": $(${SSGET} -i $i -j)}]" \
+        >"${RESULT_FILE}.imd"
+    echo -e "($i/${NUM_PROBLEMS}):\tRunning SpMV for ${GROUP}/${NAME}" 1>&2
+    ./spmv/spmv --executor=${EXECUTOR} \
+                --backup="${RESULT_FILE}.bkp" \
```
Comment (hartwiganzt) [inline benchmark/run_all_benchmarks.sh]: please explain here what is happening: write results into several files (backup) etc....

**Comment by gflegar** [comment]
Don't merge it yet, still want to do a commit.

**Comment by gflegar** [comment]
Tried to do something about the code to make it easier to understand, take a look again.

I'm still not sure what to do about the CI scripts, as I feel that them being hard to understand is just a symptom of a bigger problem: there's no good way to write reusable pieces of code (functions, macros, etc.) in the YAML file. I'll open an issue so we have a place to discuss it.

**Pull request merged**